File: ecommerce-platform-frontend/src/app/dto/product/response-product-dto.ts
import {ResponseTagDTO} from '../tag/response-tag-dto';
import {BaseUpdateOrResponseDTO} from '../base/base-update-or-response.dto';

export interface ResponseProductDTO extends BaseUpdateOrResponseDTO {
  price?: number;
  weightGrams?: number;
  categoryId?: number;
  responseTagDTOS?: ResponseTagDTO[];
  mixable: boolean;
  displayInProducts: boolean;
}

File: catalog-service/src/main/java/com/dvoracekmartin/catalogservice/application/dto/product/ResponseProductDTO.java
package com.dvoracekmartin.catalogservice.application.dto.product;

import com.dvoracekmartin.catalogservice.application.dto.media.MediaDTO;
import com.dvoracekmartin.catalogservice.application.dto.tag.ResponseTagDTO;

import java.math.BigDecimal;
import java.util.List;

public record ResponseProductDTO(
        Long id,
        String name,
        String description,
        BigDecimal price,
        List<String> images,
        Long categoryId,
        String scentProfile,
        String botanicalName,
        String extractionMethod,
        String origin,
        String usageInstructions,
        Integer volumeMl,
        String warnings,
        String medicinalUse,
        Double weightGrams,
        List<String> allergens,
        List<ResponseTagDTO> tagsDTOs,
        List<MediaDTO> mediaDTOS
) {
}
